### PR TITLE
Added click feedback

### PR DESCRIPTION
**Pull request by Nitwel**
GIF 14 02 2020 17-14-51

### CODE REVIEW

File: src/components/v-button/v-button.vue [hunk 1]
```diff
@@ -52,9 +52,17 @@ export default createComponent({
...
-		return { sizeClass };
+
+		var timeout: number = -1;
+
+		const clicked = function() {
+			if (props.loading) return;
+			emit('click');
+		};
+
+		return { sizeClass, clicked };
```
Comment (rijkvanzanten) [inline src/components/v-button/v-button.vue]: Lets change this to a function to be more in style with the other components and pass on the event:

```js
// ...
return { sizeClass, onClick };

function onClick(event: MouseEvent) {
    if (props.loading === true) return;
    emit('click', event);
}
// ...
```

File: src/components/v-button/v-button.vue [hunk 2]
```diff
@@ -52,9 +52,17 @@ export default createComponent({
 		},
 		...sizeProps
 	},
-	setup(props) {
+	setup(props, { emit }) {
 		const sizeClass = useSizeClass(props);
-		return { sizeClass };
+
+		var timeout: number = -1;
```
Comment (rijkvanzanten) [inline src/components/v-button/v-button.vue]: Where is this used? (Also use `let` / `const`)

**Comment by rijkvanzanten** [comment]
We'll probably also have to ad an additional test to handle the onClick function

File: src/components/v-button/v-button.vue (hunk 2 above)
Comment (Nitwel) [reply src/components/v-button/v-button.vue]: Remainder of testing.

File: src/components/v-button/v-button.vue [hunk 3]
```diff
@@ -4,7 +4,7 @@
 		:class="[sizeClass, { block, rounded, icon, outlined, loading }]"
 		:type="type"
 		:disabled="disabled"
-		@click="!loading ? $emit('click') : null"
+		@click="onClick()"
```
Comment (rijkvanzanten) [inline src/components/v-button/v-button.vue]:
```suggestion
		@click="onClick"
```

**Pull request merged**